src/Toml/Type/AnyValue.hs

```diff
@@ -82,8 +110,23 @@ reifyAnyValues _ []                 = Right []
 reifyAnyValues v (AnyValue av : xs) = sameValue v av >>= \Refl -> (av :) <$> reifyAnyValues v xs
 
 -- | Function for creating 'Array' from list of 'AnyValue'.
-toMArray :: [AnyValue] -> Maybe (Value 'TArray)
-toMArray [] = Just $ Array []
+toMArray :: [AnyValue] -> Either MatchError (Value 'TArray)
+toMArray [] = Right $ Array []
 toMArray (AnyValue x : xs) = case reifyAnyValues x xs of
-    Left _     -> Nothing
-    Right vals -> Just $ Array (x : vals)
+    Left _     -> leftMatchError x
+    Right vals -> Right $ Array (x : vals)
+
+----------------------------------------------------------------------
+-- Useful functions
+----------------------------------------------------------------------
+
+tShow :: Show a => a -> Text
+tShow = T.pack . show
+
+-- | Expacted value
+typeName :: forall a . Typeable a => TValue
+typeName = read . show . typeRep $ Proxy @a
```

Comment:
This function is kinda unsafe. As I understand, you want to convert type-level `TValue` tag to `Text`. In that case it will be safer to have `forall (t :: TValue)` and to use `readMaybe` returning `<unknown>` on `Nothing`.